addons/mass_mailing/controllers/main.py

```diff
@@ -161,8 +180,10 @@ def send_feedback(self, mailing_id, res_id, email, feedback, token):
             if not self._valid_unsubscribe_token(mailing_id, res_id, email, token):
                 return 'unauthorized'
             model = request.env[mailing.mailing_model_real]
-            records = model.sudo().search([('email_normalized', '=', tools.email_normalize(email))])
-            for record in records:
-                record.sudo().message_post(body=_("Feedback from %s: %s" % (email, feedback)))
-            return bool(records)
+            if model._name == 'mailing.contact':
+                records = model.sudo().search([('email_normalized', '=', tools.email_normalize(email))])
+                subscription_ids = records.mapped('subscription_list_ids').filtered(lambda x: x.opt_out and x.list_id.id in mailing.contact_list_ids.ids)
+                subscription_ids.with_user(request.env.ref('base.user_root')).write({'opt_out_reason': feedback.get('key'),
+                                                                                     'opt_out_custom': feedback.get('reason') if feedback.get('key') == 'other' else False})
+            return True
```

Comment:
you should return True only if feedback have been sent. If you cannot find any record matching the email, return False.
Also, write 
```
subscription_ids.with_user(request.env.ref('base.user_root')).write({
   'opt_out_reason': feedback.get('key'),
   'opt_out_custom': feedback.get('reason') if feedback.get('key') == 'other' else False
})
```
we could also add other return key : 'notImplemented' to tell that the target model is incorrect if model != mailing contact. (adapt caller to handle this error).
```
if model._name != 'mailing.contact':
    return 'notImplemented'
records = ...
...
return bool(subscription_ids)
```